src/simulation/dynamics/stateSpaceEffector/stateSpaceStateEffector.cpp

```diff
@@ -0,0 +1,170 @@
+    this->AEff = Eigen::MatrixXd::Zero(sizeOfStateSpace, 3);
+    this->BEff = Eigen::MatrixXd::Zero(sizeOfStateSpace, 3);
+    this->CEff = Eigen::VectorXd::Zero(sizeOfStateSpace);
+
+    // Zero the mass props and mass prop rates contributions
+    this->effProps.mEff = 0.0;
+    this->effProps.rEff_CB_B.fill(0.0);
+    this->effProps.IEffPntB_B.fill(0.0);
+    this->effProps.rEffPrime_CB_B.fill(0.0);
+    this->effProps.IEffPrimePntB_B.fill(0.0);
+
+    this->nameOfXState = "X" + std::to_string(StateSpaceStateEffector::effectorID);
+    this->nameOfXDotState = "XDot" + std::to_string(StateSpaceStateEffector::effectorID);
+    StateSpaceStateEffector::effectorID++;
+}
+
+uint64_t StateSpaceStateEffector::effectorID = 1;
+
+StateSpaceStateEffector::~StateSpaceStateEffector()
+{
+    StateSpaceStateEffector::effectorID --;    /* reset the panel ID*/
+}
+
+void StateSpaceStateEffector::Reset(uint64_t CurrentClock)
+{
+}
+
+void StateSpaceStateEffector::prependSpacecraftNameToStates()
+{
+    this->nameOfXState = this->nameOfSpacecraftAttachedTo + this->nameOfXState;
+    this->nameOfXDotState = this->nameOfSpacecraftAttachedTo + this->nameOfXDotState;
+}
+
+void StateSpaceStateEffector::linkInStates(DynParamManager& statesIn)
+{
+    this->m_SC = statesIn.getPropertyReference(this->nameOfSpacecraftAttachedTo + "m_SC");
+    this->c_B = statesIn.getPropertyReference(this->nameOfSpacecraftAttachedTo + "centerOfMassSC");
+    this->cPrime_B = statesIn.getPropertyReference(this->nameOfSpacecraftAttachedTo + "centerOfMassPrimeSC");
+    this->inertialPositionProperty = statesIn.getPropertyReference(this->nameOfSpacecraftAttachedTo + "r_BN_N");
+    this->inertialVelocityProperty = statesIn.getPropertyReference(this->nameOfSpacecraftAttachedTo + "v_BN_N");
+}
+
+void StateSpaceStateEffector::registerStates(DynParamManager& states)
+{
+    this->XState = states.registerState(this->sizeOfStateSpace, 1, this->nameOfXState);
+    this->XDotState = states.registerState(this->sizeOfStateSpace, 1, this->nameOfXDotState);
+
+    this->XState->setState(this->XInit);
+    this->XDotState->setState(this->XDotInit);
+}
+
+void StateSpaceStateEffector::updateEffectorMassProps(double integTime)
+{
+    this->X = this->XState->getState();
+    this->XDot = this->XDotState->getState();
+
+    Eigen::Vector3d r_SF_F = this->X.head(3);
+    Eigen::Vector3d r_ScB_B = this->dcm_FB.transpose() * (this->r_ScS_F + r_SF_F) + this->r_FB_B;
+    Eigen::Matrix3d rTilde_ScB_B = eigenTilde(r_ScB_B);
+    Eigen::Matrix3d ISPntB_B = this->dcm_FB.transpose() * this->ISPntSc_F * this->dcm_FB
+            - this->mass * rTilde_ScB_B * rTilde_ScB_B;
+
+    this->effProps.mEff = this->mass;
+    this->effProps.rEff_CB_B = r_ScB_B;
+    this->effProps.IEffPntB_B = ISPntB_B;
+}
+
+void StateSpaceStateEffector::updateContributions(double integTime, BackSubMatrices & backSubContr, Eigen::Vector3d sigma_BN, Eigen::Vector3d omega_BN_B, Eigen::Vector3d g_N)
```

Comment:
:large_orange_diamond: Can drop each parameter down on a new line so the line is within 120 characters.

Reply:
Done.